README.md

```diff
@@ -238,7 +243,8 @@ const query = useInfiniteAPIQuery(
 
 The return value of this hook is identical to the behavior of the `react-query` `useInfiniteQuery` hook's return value where `data` holds an array of pages.
 
-When returning `undefined` from `getNextPageParam` it will set `query.hasNextPage` to false, otherwise it will merge the next api request payload with the returned object, likewise for `getPreviousPageParam` and `query.hasPreviousPage`.
+When returning `undefined` from `getNextPageParam` it will set `query.hasNextPage` to false, otherwise it will merge the next api request payload with the returned object, likewise for `getPreviousPageParam` and `query.hasPreviousPage`. This is useful to pass pagination token from previous page since the current implementation provides a default `queryFn` assumes such token
+is required over query string. It may need another queryFn if the pagination token is managed via headers.
```

Comment:
do we have an api that does pagination through headers? wow thats crazy to me!

Reply:
No from openapi. Since `@tanstack/query` is spec agnostic and this is an OSS repo, it's better to highlight the assumption on pagination design.

Reply:
@jkdowdle If you're interested in an example, the GitHub API often uses headers for pagination.

https://docs.github.com/en/rest/guides/using-pagination-in-the-rest-api?apiVersion=2022-11-28